aws/cluster/main.tf

```diff
@@ -113,3 +113,18 @@ EOF
 
   depends_on = [null_resource.kops-cluster]
 }
+
+// Cluster Docker Login Configuration on S3
+// Short version of https://github.com/kubernetes/kops/blob/master/docs/cli/kops_create_secret_dockerconfig.md
+// Kops version currently subject to bugs and doesn't trigger rolling upgrade so no need to complicate as per cluster-spec.
+resource "aws_s3_bucket_object" "docker-auth-config" {
+  count  = var.docker-auth-config != "" ? 1 : 0
+  bucket = var.kops-state-bucket
+  key    = "/${var.cluster-name}/secrets/dockerconfig"
```

Comment:
Are the files from `secrets` folder picked up automatically, or do we need a local-exec provisioner here?

Reply:
They are picked up automatically by new nodes. The older nodes need manual rotation using kops as there is a bug causing kops to not try and do this automatically.